nixos/modules/system/boot/loader/systemd-boot/systemd-boot.nix

```diff
@@ -280,7 +277,7 @@ in {
       # be updated to use the 32-bit EFI app on 32-bit systems.  The 32-bit EFI
       # app filename is BOOTIA32.efi.
       (mkIf cfg.memtest86.enable {
-        "efi/memtest86/BOOTX64.efi" = "${pkgs.memtest86-efi}/BOOTX64.efi";
+        "efi/memtest86/memtest.efi" = "${pkgs.memtest86plus.efi}";
```

Comment:
It seems like the package expose a `.efi` passthrough, it would be good to have a symmetric code for GRUB by having a `legacy` and `hybrid` passthrough for the other possibilities.

Reply:
Got it

Reply:
(Note about resolution, discussed about it in the matrix room, and it is not necessary to deal with the 32 bit path since it is intrinsically handled by the i686 system being built)